### DISCUSSION TITLE
[CONTRACTS] - OHM passed a billion TVL

### DISCUSSION THREAD

**Discussion by 0x4007** [pavlovcik/uad-ui#134] [Ideas]
[What is Olympus Pro? | Can bonds really replace liquidity mining?](https://www.youtube.com/watch?v=mGiRzFIw9Ds)
Olympus is a decentralized reserve currency protocol based on the OHM token. Each OHM token is backed by a basket of assets (e.g. DAI, FRAX) in the Olympus treasury, giving it an intrinsic value that it cannot fall below. 

[https://www.olympusdao.finance/](https://www.olympusdao.finance/)

I wonder if any of those ponzinomics can be feasibly applied to the Ubiquity Dollar? Makes me wonder if we should make uBOND be redeemable EITHER when TWAP above 1.00 or automatically after x amount of days?

**Comment by 0x4007** [comment]
Apparently there's some consensus around that OHM is basically ESD with a treasury of actual stable stables. Seems like this bodes well for us. @hpiyankov still curious on your thoughts about OHM and how we can basically extract the best elements from it because of its wild success.


[How is OHM model different from algo stable models](https://t.me/lobsters_chat/275303)
> ["How is this not ESD but just with treasury being in non-shit stable?"](https://t.me/lobsters_chat/275308)

**Comment by hpiyankov** [comment]
@pavlovcik apparently I am missing something here. Let's go over how OHM works

1. For every OHM minted there is at least 1 DAI (or another stable) in the treasury (theoretically). Currently:
- there is 2.1 mil OHM in circulation https://app.olympusdao.finance/#/dashboard
- there is 187 mil USD in their treasury
- I don't know where the TVL comes from, but I guess it is 2.1 mil OHM * 677 prices, so apparently most of the OHM is staked

2. You can bond, which means that you provide liquidity to OHM and then using the LP tokens you buy OHM at a discount, at a later stage. An important point here is that you lose your LP stake. This is de-facto a purchase.

3. You can stake OHM which decreases circulation and in exchange, you receive more OHM when it is being minted. 

So putting all of the above together you get to:
1. Buy ohm (increase price) > Provide liquidity (stabilize price)  > Recieve OHM at a discount > Repeat / Sell at a small profit / Stake
2. Stake ohm > receive seigniorage when supply is expanded

At the end of the day, there are still just 2.1 mil OHM in circulation, which means that these is all the reserves which they need. They are defacto over collateralised 100x right now.

I don't get how this is anything similar to ESD? OHM is not an algo-coin it is a collateralised stable coin with 100% collateral.... in other stables.

**Reply by 0x4007** [reply]
> I don't get how this is anything similar to ESD? OHM is not an algo-coin it is a collateralised stable coin with 100% collateral.... in other stables.

I'm just forwarding what they said:

> "[I tried to understand ohm thrice now and still think it's just a slower rebasing ESD.](https://t.me/lobsters_chat/275330)"

> "[It is ESD but with treasury. Keeping the peg never worked so they just shifted torwards accumulation of stables as long term goal.](https://t.me/lobsters_chat/275336)"

When asked "[How it is different from other rebasing tokens but with extremely slow rebases and a tiny collateralization ratio?](https://twitter.com/0xdjma/status/1417278642829283331?s=20)" Zues, the creator of OHM replied "[biggest difference imo is no peg, and thus no (or at least fewer) existential threats that signal to the market that its time to pour out](https://twitter.com/ohmzeus/status/1417280808285609984)"

**Reply by hpiyankov** [reply]
This is word play. When they so no peg, they mean "no ceil on the price". They do have a "price floor" though. 

Anyway, OHM is 100% collateralised stablecoin, which is quite far from what we do. Regarding their tokenomics.... a lot of people will be very disappointed when the APY spiral start to unfurl. 

To address your question more directly - it has success because it is ponzi-like. I am not sure we want to extract this.

**Comment by 0x4007** [comment]
https://olympusdao.medium.com/introducing-olympus-pro-d8db3052fca5

**Reply by hpiyankov** [reply]
Yes, this is in a nutshell what I described on our lst call: https://docs.olympusdao.finance/pro/olympus-pro-users/what-are-bonds

User A creates an LP position
User A sells the LP position to the protocol at a premium in exchange for the protocol's native tokens
The tokens which the user gets from the sale vest/have a cliff
The protocol controls the discount at which the tokens are purcahsed

**Reply by 0x4007** [reply]
So basically this is exactly the deal we have already been giving our pre/seed investors - that they lock up liquidity for four years, and we give them debt tokens at a premium. The only difference being that OHM has a fixed maturation while ours is based on when the TWAP goes above 1.00.

Makes me want to automate this functionality and market it as OHM 2.0 😛

**Reply by hpiyankov** [reply]
It is different. OHM purchases the liquidity outright, while at the end of our period, investors can withdraw it.

**Reply by 0x4007** [reply]
Should we purchase the liquidity outright then? The only issue with outright purchase is that they wouldn't have a chance at accruing seignorage or farming rewards.

I feel like it would be fair to give a smaller premium upfront because the users can receive UBQ farming + (their principle & seignorage at the end of the lockup.) Seignorage is a good long term sustainable yield method, but I fear that the apes would rather optimize for upfront money, meaning, for growth reasons I'm leaning towards outright purchase with higher rewards upfront- then when winter hits we do the more "sustainable" method?

**Reply by hpiyankov** [reply]
The outright purchase has huge implications on the overall token economy for multiple reasons. If we go in this direction, this is defacto a significant overhaul of the protocol (I am trying not to say a full overhaul).
